explorer/src/components/JupiterSwapModal.tsx

```diff
@@ -0,0 +1,562 @@
+	const wallet = useWallet();
+
+	const [allTokens, setAllTokens] = useState<Token[]>([]);
+	const [tokens, setTokens] = useState<Token[]>([]);
+	const [tokenMap, setTokenMap] = useState<Map<string, TokenInfo>>(new Map());
+	const [inputToken, setInputToken] = useState<Token>({
+		chainId: 0,
+		address: '',
+		symbol: '',
+		name: '',
+		decimals: 3,
+		logoURI: '',
+		tags: [''],
+	});
+	const [outputToken, setOutputToken] = useState<Token>({
+		chainId: 0,
+		address: '',
+		symbol: '',
+		name: '',
+		decimals: 3,
+		logoURI: '',
+		tags: [''],
+	});
+	const [tokenSearchType, setTokenSearchType] = useState<string>("");
+	const [exchangeAmount, setExchangeAmount] = useState<number>(1);
+	const [displayRoutes, setDisplayRoutes] = useState<any[]>([]);
+	const [additionalRoutes, setAdditionalRoutes] = useState<any[]>([]);
+	const [tokenLoading, setTokenLoading] = useState<boolean>(false);
+
+	const [balance, setBalance] = useState(0);
+	const [recieveTokenBalance, setRecieveTokenBalance] = useState(0);
+	const [inputMint, setInputMint] = useState<PublicKey | undefined>(undefined);
+	const [outputMint, setOutputMint] = useState<PublicKey | undefined>(undefined);
+	const [showTokenSearch, setShowTokenSearch] = useState<boolean>(false);
+	const [tokenSearch, setTokenSearch] = useState("");
+
+	const [formValue, setFormValue] = useState<UseJupiterProps>({
+		amount: 1 * 10 ** 6,
+		inputMint: inputMint ? new PublicKey(inputMint) : undefined,
+		outputMint: outputMint ? new PublicKey(outputMint) : undefined,
+		slippage: 0.1,
+	});
+	const [inputBalance, setInputBalance] = useState(0);
+
+	const [showSlippageSettings, setShowSlippageSettings] =
+		useState<boolean>(false);
+	const [selectSlippage, setSelectSlippage] = useState<number | null>(null);
+	const [inputSlippage, setInputSlippage] = useState("");
+
+	const [showAggregatorSearch, setShowAggregatorSearch] =
+		useState<boolean>(false);
+	const [aggregator, setAggregator] = useState<Aggregator | null>(null);
+	const [aggregatorOptions, setAggregatorOptions] = useState<Array<Aggregator>>(aggregators);
+	const [aggregatorSearch, setAggregatorSearch] = useState("");
+
+	const [showMore, setShowMore] = useState(false);
+
+	const setWalletBalance = async (tokenAddress: string, type: string) => {
+		if (wallet?.publicKey) {
+			try {
+				const info = await connection.getParsedTokenAccountsByOwner(wallet.publicKey, {
+					mint: new PublicKey(tokenAddress),
+				},
+				);
+				const balance = info.value[0].account.data.parsed.info.tokenAmount.uiAmount;
+				if (type === "pay") {
+					setBalance(balance);
+					if ((balance) - 0.00001 > 0) {
+						setInputBalance(balance - 0.00001);
+					} else {
+						setInputBalance(0);
+					}
+				} else if (type === "receive") {
+					setRecieveTokenBalance(balance);
+				}
+			} catch (error) {
+				console.log(error);
+			}
+		}
+	}
+
+	useEffect(() => {
+		setTokenLoading(true);
+		fetch(TOKEN_LIST_URL["mainnet-beta"])
```

Comment:
Allow token list for users to select to be dynamic (based on their current network)